tap_kafka/sync.py

```diff
@@ -61,9 +62,41 @@ def update_bookmark(state, topic, message):
                                  })
 
 
+def initial_start_time_to_offset_reset(initial_start_time: str) -> str:
+    """Convert initial_start_time to corresponding kafka auto_offset_reset"""
+    return 'earliest' if initial_start_time == 'earliest' else 'latest'
+
+
+def iso_timestamp_to_epoch(iso_timestamp: str) -> int:
+    """Convert an ISO 8601 formatted string to epoch in milliseconds"""
+    try:
+        return int(dateutil.parser.parse(iso_timestamp).timestamp() * 1000)
+    except dateutil.parser.ParserError:
+        raise InvalidTimestampException(f'{iso_timestamp} is not a valid ISO formatted string.')
+
+
+def assign_consumer(consumer, topic: str, state: dict, initial_start_time: str) -> None:
+    """Assign consumer to the right position where we need to start consuming data from
+
+    If state is empty and the initial_start_time is a timestamp and not a reserved word
+    then assign each partition to the offsets at the provided timestamp.
+
+    If state exists then assign each partition to the positions in the state
+
+    If both state and initial_start_time are empty then should not assign"""
+    if not state and not initial_start_time:
+        return
+
+    if not state and initial_start_time not in ['latest', 'earliest']:
```

Comment:
Why `initial_start_time not in ['latest', 'earliest']`?

Reply:
if it's one of the reserved `latest` or `earliest` words then we should not try to assigned to consumer to specific timestamp. In that case we need to get bookmarked values.

Reply:
hmm, what happens in the case we don't have a state but `initial_start_time` is either `latest` or `earliest`, as per this `if-else` block , it would use the bookmarks but we don't have any

Reply:
same as above, d741a00d04886161b771cf05490f3a6f21996f82 simplifies these conditions. Can you please take a look again?